ironmon_tracker/screens/ExtrasScreen.lua

```diff
@@ -171,5 +231,39 @@ function ExtrasScreen.drawScreen()
 	end
 
 	local ivBtn = ExtrasScreen.Buttons.EstimateIVs
-	Drawing.drawText(topboxX + 4, ivBtn.box[2] + ivBtn.box[4] + 1, ivBtn.ivText, Theme.COLORS[ivBtn.textColor], shadowcolor)
-end
\ No newline at end of file
+	if ivBtn then
+		Drawing.drawText(topboxX + 4, ivBtn.box[2] + ivBtn.box[4] + 1, ivBtn.ivText, Theme.COLORS[ivBtn.textColor],
+		shadowcolor)
+	end
+end
+
+function ExtrasScreen.viewPreviousLogFile()
+	local romname, rompath
+	if Options["Use premade ROMs"] and Options.FILES["ROMs Folder"] ~= nil then
+		-- First make sure the ROMs Folder ends with a slash
+		if Options.FILES["ROMs Folder"]:sub(-1) ~= FileManager.slash then
+			Options.FILES["ROMs Folder"] = Options.FILES["ROMs Folder"] .. FileManager.slash
+		end
+
+		romname = GameSettings.getRomName() or ""
+		rompath = Options.FILES["ROMs Folder"] .. romname .. FileManager.Extensions.GBA_ROM
+		if not FileManager.fileExists(rompath) then
+			romname = romname:gsub(" ", "_")
+			rompath = Options.FILES["ROMs Folder"] .. romname .. FileManager.Extensions.GBA_ROM
+		end
+	elseif Options["Generate ROM each time"] then
+		-- Filename of the AutoRandomized ROM is based on the settings file (for cases of playing Kaizo + Survival + Others)
+		local quickloadFiles = Main.GetQuickloadFiles()
+		local settingsFileName = FileManager.extractFileNameFromPath(quickloadFiles.settingsList[1] or "")
+		romname = string.format("%s %s%s", settingsFileName, FileManager.PostFixes.PREVIOUSATTEMPT,
+		FileManager.Extensions.GBA_ROM)
+		rompath = FileManager.prependDir(romname)
+	end
+
+	local logpath = FileManager.getPathIfExists((rompath or "") .. FileManager.Extensions.RANDOMIZER_LOGFILE)
+	if logpath == nil then
+		return false
+	end
+
+	return LogOverlay.parseAndDisplay(logpath)
+end
```

Comment:
As this function looks more or less identical to `GameOverScreen.viewLogFile()`, I think it would be best to re-use that function here. Provide a parameter for the function `postfix` to provide the choice to replace `FileManager.PostFixes.AUTORANDOMIZED` and `FileManager.PostFixes.PREVIOUSATTEMPT`. Then update the areas of the code that call the function in question with the appropriate parameter, for example:

`GameOverScreen.viewLogFile(FileManager.PostFixes.AUTORANDOMIZED)`